yetanotherco/aligned_layer
# feat(verifier) Add nexus verifier

closes #1159 

Currently the batcher fails to compile due to a dependency discrepancy between `risc0-zkvm` and `nexus-core` relying on different version of `ark-groth16`. To observe this comment the crate declaration for [risc0](https://github.com/yetanotherco/aligned_layer/pull/1175/files#diff-7745fc91d312f3c1dae24301ac51b757ded7a152052adafa9c78851ffccf9f8eR29) in batcher/Cargo.toml and run `cargo build`. You should observe that the batcher compiles but errors due to the risc0 dependency not being present.
<img width="705" alt="Screenshot 2024-10-04 at 04 02 16" src="https://github.com/user-attachments/assets/77a66f62-5c19-4660-80f6-25ca9f9e867c">

Note: When previously testing this intengration the msg size of the websocket was not large enough to process some proofs: https://github.com/yetanotherco/aligned_layer/pull/1175/files#diff-fb54942eba81c272160ef10dfb2972fd0a6de56d45ed87c98ca1a65b3b2dd29bR386.

## To Test:
- Run `make test_nexus_rust_ffi` and `make test_nexus_go_bindings_macos`/`make test_nexus_go_bindings_linux`
- Run local testnet and send a Nexus proofs to the batcher  by running `make batcher_send_nexus_burst` and confirm they are accpeted.
